build-test.sh

```diff
@@ -0,0 +1,653 @@
+        fi
+    fi
+}
+
+build_Tests()
+{
+    __TestDir=$__ProjectDir/tests
+    __ProjectFilesDir=$__TestDir
+    __TestBinDir=$__TestWorkingDir
+
+    if [ $__RebuildTests -ne 0 ]; then
+        if [ -d "${__TestBinDir}" ]; then
+            echo "Removing tests build dir: ${__TestBinDir}"
+            rm -rf $__TestBinDir
+        fi
+    fi
+
+    __CMakeBinDir="${__TestBinDir}"
+
+    if [ -z "$__TestIntermediateDir" ]; then
+        __TestIntermediateDir="tests/obj/${__BuildOS}.${__BuildArch}.${__BuildType}"
+    fi
+
+	echo "__BuildOS: ${__BuildOS}"
+	echo "__BuildArch: ${__BuildArch}"
+	echo "__BuildType: ${__BuildType}"
+	echo "__TestIntermediateDir: ${__TestIntermediateDir}"
+
+    if [ ! -f "$__TestBinDir" ]; then
+        echo "Creating TestBinDir: ${__TestBinDir}"
+        mkdir -p $__TestBinDir
+    fi
+    if [ ! -f "$__LogsDir" ]; then
+        echo "Creating LogsDir: ${__LogsDir}"
+        mkdir -p $__LogsDir
+    fi
+
+    __BuildProperties="-p:OSGroup=${__BuildOS} -p:BuildOS=${__BuildOS} -p:BuildArch=${__BuildArch} -p:BuildType=${__BuildType}"
+
+    # =========================================================================================
+    # ===
+    # === Restore product binaries from packages
+    # ===
+    # =========================================================================================
+
+    if [ -z "$XunitTestBinBase"]; then
+      XunitTestBinBase="$__TestWorkingDir"
+    fi
+
+    export CORE_ROOT="$XunitTestBinBase/Tests/Core_Root"
+    CORE_OVERLAY="$XunitTestBinBase/Tests/Core_Root_$__RuntimeId"
+
+    if [ ! -f "${CORE_ROOT}" ]; then
+        mkdir -p ${CORE_ROOT}
+    else
+        rm -rf ${CORE_ROOT}
+    fi
+
+    cp -r $__BinDir/* $CORE_ROOT/ > /dev/null
+
+    # make sure the correct ilasm will be used
+    export CscToolPath="${CORE_ROOT}"
+
+    build_Tests_internal "Restore_Product" "${__ProjectDir}/tests/build.proj" " -UpdateDependencies -BatchRestorePackages  ${__BuildAgainstPackagesArg}" "Restore product binaries (build tests)"
+
+    build_Tests_internal "Tests_GenerateRuntimeLayout" "${__ProjectDir}/tests/runtest.proj" "-BinPlaceRef -BinPlaceProduct -CopyCrossgenToProduct ${__BuildAgainstPackagesArg}" "Restore product binaries (run tests)"
+
+    if [ -n "$__UpdateInvalidPackagesArg" ]; then
+        __up=-updateinvalidpackageversion
+    fi
+
+    # Work hardcoded path around
+    if [ ! -f "${__BuildToolsDir}/Microsoft.CSharp.Core.Targets" ]; then
+        ln -s "${__BuildToolsDir}/Microsoft.CSharp.Core.targets" "${__BuildToolsDir}/Microsoft.CSharp.Core.Targets"
+    fi
+    if [ ! -f "${__BuildToolsDir}/Microsoft.CSharp.targets" ]; then
+        ln -s "${__BuildToolsDir}/Microsoft.CSharp.Targets" "${__BuildToolsDir}/Microsoft.CSharp.targets"
+    fi
+
+    echo "Starting the Managed Tests Build..."
+
+    __ManagedTestBuiltMarker=${__TestBinDir}/managed_test_build
+
+    if [ ! -f $__ManagedTestBuiltMarker ]; then
+
+        __BuildAgainstPackagesArg=-BuildTestsAgainstPackages
```

Comment:
This & other instances of setting "__BuildAgainstPackagesArg" should be removed if you implement the above fix